robottelo/datafactory.py

```diff
@@ -66,6 +67,33 @@ def generate_strings_list(length=None, exclude_types=None, bug_id=None,
     return list(strings.values())
 
 
+def fix_string_to_include_uppercase(st):
```

Comment:
too redundant
```
def add_uppercase_char_into_string(str=None, length=10)
if str is None:
    str = gen_string('alpha', length)
st_chars = list(str)
if not any(char in string.ascii_uppercase for char in st_chars):
    st_chars[random.randint(0, len(st_chars)-1)] = random.choice(
        string.ascii_uppercase)
    return ''.join(st_chars)
else:
    return str
```

suggest this

if person wants generate string of non-default type (which is unlikeable) with uppercase, I think it is fine to:
```
a = gen_string('utf8')
b = add_uppercase_char_into_string(a)
```

That is only my opinion and it will be nice to hear other people thougths

Reply:
Just a note, if "str" is a unicode, we cannot check  upper case with:
```python
if not any(char in string.ascii_uppercase for char in st_chars):  
```
the simplest (with unicode) is to convert each char to upper case and compare equality with initial one 
```python
a = u'пРивет'
for che in a:
    print '>>', che, che.upper() == che
             
>> п False
>> Р True
>> и False
>> в False
>> е False
>> т False

# but with string.ascii_uppercase            
for che in a:
    print '>>', che, che.upper() in string.ascii_uppercase            
>> п False
>> Р False
>> и False
>> в False
>> е False
>> т False


```

Reply:
eg use:
```python
any(char == char.upper() for char in list(a))

```

Reply:
there is also a problem with special chars and unicode that must be resolved, so perhaps to simplify this function do not support unicode will be more simple

Reply:
for any string unicode or not if str == str.lower() mean that str contain no uppercase

Reply:
think also that one function should be sufficient